pkg/sql/catalog/descs/collection.go

```diff
@@ -1324,6 +1324,42 @@ func (tc *Collection) GetConstraintComment(
 	return tc.GetComment(catalogkeys.MakeCommentKey(uint32(tableID), uint32(constraintID), catalogkeys.ConstraintCommentType))
 }
 
+// ErrDescCannotBeLeased indicates that the full Get method is needed to
+// lock this descriptor. This can happen if synthetic descriptor or uncommitted
+// descriptors are in play. Or if the leasing layer cannot satisfy the request.
+var ErrDescCannotBeLeased = errors.New("descriptor cannot be leased")
```

Comment:
nit: can we include the desc ID in these error messages? i know the error isn't used anywhere right now, but it might be in the future.